googletest/include/gtest/internal/gtest-internal.h

```diff
@@ -1337,17 +1346,24 @@ constexpr bool InstantiateTypedTestCase_P_IsDeprecated() { return true; }
 
 #define GTEST_TEST_NO_THROW_(statement, fail) \
   GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
-  if (::testing::internal::AlwaysTrue()) { \
+  if (::testing::internal::TrueWithString gtest_msg = std::string()) { \
     try { \
       GTEST_SUPPRESS_UNREACHABLE_CODE_WARNING_BELOW_(statement); \
     } \
+    catch (std::exception const& e) { \
+      gtest_msg.value = " std::exception-derived exception with description: "; \
+      gtest_msg.value += e.what(); \
+      gtest_msg.value += "."; \
```

Comment:
The value of `what()` is implementation defined. Why does adding a dot make sense?

Reply:
To make sure we do have a dot at the end of the sentence, as we have in case of any other exception.

Reply:
Instead of adding a period at the end, are you able to change this such that the output is like the following?

```
gtest_msg.value = "`std::exception`-derived exception thrown with description \""
                                + e.what()
                                "\"";
```

Reply:
That works, for me! Add the quotes around the error message and then keep the dot as well :)

Reply:
updated